aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -33,6 +34,15 @@ def after_initialize(client)
         end
       end
 
+      private
+      def self.resolve_region(cfg)
+        keys = %w(AWS_REGION AMAZON_REGION AWS_DEFAULT_REGION)
+        env_region = ENV.values_at(*keys).compact.first
+        env_region = nil if env_region == ''
```

Comment:
Good catch on the empty ENV region string. I don't believe we check for this previously.